### PR TITLE
Allow extracted semantics tests to run with and without yul.

### PR DESCRIPTION
**Pull request by ekpyron**
The need for this came up in #6888 and #6909.
This is a quick and dirty solution which we can approve upon later, but which will allow us to continue with #6888 soon.

### CODE REVIEW

**Comment by ekpyron** [comment]
This let's us define ``compileViaYul: also``, which lets the test be run with and without yul.

It's first run without yul as usual. In case of expectation mismatch ``isoltest`` will do the same as always and print an additional warning stating that the test has to pass via yul as well.
In case without yul passes, the test is re-run with yul. If that passes, everything's fine. If that fails, ``isoltest`` can update expectations as usual, but prints the additional error message stating that the same test *did* pass without yul.

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -46,77 +46,102 @@ SemanticTest::SemanticTest(string const& _filename, string const& _ipcPath, lang
...
+			m_runWithYul = true;
+			m_runWithoutYul = false;
+		}
 	}
 	parseExpectations(file);
 }
 
 TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePrefix, bool _formatted)
 {
+	for(bool compileViaYul: set<bool>{!m_runWithoutYul, m_runWithYul})
```
Comment (Marenz) [inline test/libsolidity/SemanticTest.cpp:68]: wouldn't this run ALL tests twice?

Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp:68]: Cases:
- ``m_runWithoutYul = false, m_runWithYul = false``:
   cannot happen
- ``m_runWithoutYul = true, m_runWithYul = false``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {false, false} = {false}``
   => runs only with ``compileViaYul = false``
- ``m_runWithoutYul = false, m_runWithYul = true``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {true, true} = {true}``
   => runs only with ``compileViaYul = true``
- ``m_runWithoutYul = true, m_runWithYul = true``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {false,true} ``
   => runs with ``compileViaYul = false`` and with ``compileViaYul = true``

Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp:68]: Or if it's not the values you were concerned about, but their multiplicity: it's a ``std::set`` - it'll never contain the same value twice.

File: test/libsolidity/SemanticTest.cpp [hunk 2]
```diff
@@ -46,77 +46,102 @@ SemanticTest::SemanticTest(string const& _filename, string const& _ipcPath, lang
...
-		m_compileViaYul = true;
-		m_settings.erase("compileViaYul");
+		if (m_settings["compileViaYul"] == "also")
+		{
+			m_validatedSettings["compileViaYul"] = m_settings["compileViaYul"];
+			m_runWithYul = true;
+			m_runWithoutYul = true;
+		}
+		else
+		{
```
Comment (leonardoalt) [inline test/libsolidity/SemanticTest.cpp:56]: Could assert that `m_settings["compileViaYul"] == "only"`

Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp:56]: Then we'd need to update all of them now, though :-)... my plan was to postpone that until we're sure we stay with this syntax... (I'm by the way open for alternative suggestions)

File: test/libsolidity/SemanticTest.cpp (hunk 1 above)
Comment (Marenz) [reply test/libsolidity/SemanticTest.cpp:68]: Right, i forgot that (main) aspect of a set. Not used seeing it in this context

File: test/libsolidity/SemanticTest.cpp [hunk 3]
```diff
@@ -46,77 +46,102 @@ SemanticTest::SemanticTest(string const& _filename, string const& _ipcPath, lang
 	m_source = parseSourceAndSettings(file);
 	if (m_settings.count("compileViaYul"))
 	{
-		m_validatedSettings["compileViaYul"] = m_settings["compileViaYul"];
-		m_compileViaYul = true;
-		m_settings.erase("compileViaYul");
```
Comment (leonardoalt) [inline test/libsolidity/SemanticTest.cpp]: Won't removing this line have any effects?

Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp]: That wasn't actually intentional... but apparently it *doesn't* have any effects :-D. I'll add it back anyways.

Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp]: Added it back.

**Review by Marenz — APPROVED** [review]

File: test/libsolidity/SemanticTest.cpp (hunk 1 above)
Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp:68]: I think it's in fact the shortest way to express this :-). But let me know if you come up with something even shorter!

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**